src/main/java/org/broadinstitute/hellbender/utils/config/GATKConfig.java

```diff
@@ -0,0 +1,146 @@
+package org.broadinstitute.hellbender.utils.config;
+
+import org.aeonbits.owner.Accessible;
+import org.aeonbits.owner.Mutable;
+import org.aeonbits.owner.Config.LoadPolicy;
+import org.aeonbits.owner.Config.LoadType;
+import org.aeonbits.owner.Config.Sources;
+
+import java.util.List;
+
+/**
+ * Configuration file for GATK options.
+ */
+@LoadPolicy(LoadType.MERGE)
+@Sources({
+        "file:${" + GATKConfig.CONFIG_FILE_VARIABLE_NAME + "}",                         // Variable for file loading
+        "file:GATKConfig.properties",                                                   // Default path
+        "classpath:org/broadinstitute/hellbender/utils/config/GATKConfig.properties"    // Class path
+})
+public interface GATKConfig extends Mutable, Accessible {
```

Comment:
Ah.  I see.

1) There are hard defaults that we've coded in the `GATKConfig` class itself.  The initial idea was to override these defaults through a config file that you'd pass in.  However, I've now added another path variable in the `Sources` annotation that comes after the file input (in terms of precedence) that will load from a variable class path.  This will allow you to include a new override in your repackaged jar without exposing the file itself to the user.

2) I think what you describe is fine to do - under the hood it's just storing the `Config` instances in a `Map` where the key is the config class and the value is the config instance.   FYI as part of refactoring to allow better downstream use, I've created some methods in `ConfigUtils` that should be used to create new configurations and retrieve cached configurations (mirroring the interface from `ConfigFactory` and `ConfigCache`).  This fixes some of the issues with loading configurations and removes the gross static block at the top of `Main` and `BaseTest`.

3) I think the class path override will work for this case, I'm less certain about the granularity of the includes.  Would an `excludeLoaders` config property be sufficient?